app/locales/languages.js

```diff
@@ -70,6 +73,40 @@ export function findUserLang(callback) {
   });
 }
 
+export async function localeChanged(locale) {
+  // If user picks manual lang, update and store setting
+  try {
+    await i18next.changeLanguage(locale);
+    await SetStoreData(LANG_OVERRIDE, locale);
+  } catch (e) {
+    console.log('something went wrong in lang change', e);
+  }
+}
+
+// This function is for checking/change current app direction
+export function languageDirectionHandler(language, navigation) {
+  // check if is not rtl and user choose rtl language
+  if (i18next.dir() !== i18next.dir(language)) {
+    showAlert(
+      null,
+      i18next.t('label.language_change_alert_title'),
+      () => [
+        localeChanged(language),
+        I18nManager.forceRTL(i18next.dir(language) === 'rtl' ? true : false),
+        ReactNativeRestart.Restart(),
```

Comment:
This restart stuff is not sitting well with me. It's like a page refresh on the web. 

* It's not going to play well with tests because the whole app restarts, losing state/context.
* There are reports of it being a **memory leak** and can end up with **crash loops**.

> Warning about using react-native-restart for this (and in general): if any problem occurs during your restart process, the app can fall into crash-restart loop that may see your app get rated as poor quality on app stores. It often works, but it isn't a good solution to this LTR<>RTL problem.

https://github.com/facebook/react-native/issues/17867#issuecomment-423858132

Can you investigate making the RTL aware components more reactive? i'm pulling in `react-i18next` in #572 which will allow better reactivity.

Reply:
Added and started the change with Typography and settingsItem, should update other components too in the furture